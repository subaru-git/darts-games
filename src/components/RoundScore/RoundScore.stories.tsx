/* eslint-disable import/named */
import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import RoundScore from './RoundScore'

export default {
  title: 'Components/RoundScore',
  component: RoundScore,
} as ComponentMeta<typeof RoundScore>

const Template: ComponentStory<typeof RoundScore> = (args) => <RoundScore {...args} />

export const Default = Template.bind({})
Default.args = {
  scores: ['19T', '0', 'D-BULL'],
}
