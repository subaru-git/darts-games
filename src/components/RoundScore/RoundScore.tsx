import React, { FC } from 'react'
import { Grid, GridItem, Heading, IconButton } from '@chakra-ui/react'
import { VscTrash } from 'react-icons/vsc'
import RoundScoreButton from '../RoundScoreButton'

type RoundScoreProps = {
  scores: string[]
  onClear: () => void
  onRoundChange: () => void
  isFinished: boolean
  onRoundOver: () => void
}

const RoundScore: FC<RoundScoreProps> = ({
  scores,
  onClear,
  onRoundChange,
  isFinished,
  onRoundOver,
}) => {
  return (
    <Grid templateColumns='repeat(4, 1fr)' alignItems='center' gap={2}>
      <GridItem>
        <Heading
          as='h2'
          size={{ base: 'md', md: 'xl' }}
          textAlign='center'
          color='gray.500'
          fontStyle='italic'
        >
          {scores[0] === '0' ? '-' : scores[0]}
        </Heading>
      </GridItem>
      <GridItem>
        <Heading
          as='h2'
          size={{ base: 'md', md: 'xl' }}
          textAlign='center'
          color='gray.500'
          fontStyle='italic'
        >
          {scores[1] === '0' ? '-' : scores[1]}
        </Heading>
      </GridItem>
      <GridItem>
        <Heading
          as='h2'
          size={{ base: 'md', md: 'xl' }}
          textAlign='center'
          color='gray.500'
          fontStyle='italic'
        >
          {scores[2] === '0' ? '-' : scores[2]}
        </Heading>
      </GridItem>
      <GridItem>
        <Grid templateColumns='repeat(2, auto)' columnGap={2}>
          <GridItem>
            <IconButton
              aria-label='clear scores'
              size={{ base: 'sm', md: 'md' }}
              icon={<VscTrash />}
              colorScheme='blue'
              onClick={() => onClear()}
              disabled={scores.length === 0}
            ></IconButton>
          </GridItem>
          <GridItem>
            <RoundScoreButton
              isFinished={isFinished}
              disabled={scores.length !== 3}
              onRoundChange={onRoundChange}
              onRoundOver={onRoundOver}
            />
          </GridItem>
        </Grid>
      </GridItem>
    </Grid>
  )
}

export default RoundScore
