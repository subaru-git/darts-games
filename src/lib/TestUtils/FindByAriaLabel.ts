export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
