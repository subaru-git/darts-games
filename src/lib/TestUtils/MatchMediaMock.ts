import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
