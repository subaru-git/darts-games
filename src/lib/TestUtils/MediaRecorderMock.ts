export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
