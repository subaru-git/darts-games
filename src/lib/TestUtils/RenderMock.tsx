import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
