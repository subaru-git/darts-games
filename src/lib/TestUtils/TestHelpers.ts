import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}