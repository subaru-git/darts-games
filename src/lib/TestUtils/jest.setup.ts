import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
