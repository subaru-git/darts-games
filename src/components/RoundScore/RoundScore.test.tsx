import { ChakraProvider } from '@chakra-ui/react';
import { fireEvent, render, screen } from '@testing-library/react';
import { createMatchMedia } from '../../lib/TestUtils/MatchMediaMock';
import RoundScore from './RoundScore';

test('should rendering', () => {
  const clear = jest.fn();
  const roundChange = jest.fn();
  const roundOver = jest.fn();
  window.matchMedia = createMatchMedia(1000);
  const { container } = render(
    <ChakraProvider>
      <RoundScore scores={['20T', '19T', '18T']} isFinished={false} result={'Result'} onClear={clear} onRoundChange={roundChange} onRoundOver={roundOver} />
    </ChakraProvider>,
  );
  expect(screen.getAllByText('20T')).toHaveLength(1);
  expect(screen.getAllByText('19T')).toHaveLength(1);
  expect(screen.getAllByText('18T')).toHaveLength(1);
  fireEvent.click(screen.getByLabelText('clear scores'));
  expect(clear).toBeCalledTimes(1);
  fireEvent.click(screen.getByText('Round Change'));
  expect(roundChange).toBeCalledTimes(1);
  expect(container).toMatchSnapshot();
});
