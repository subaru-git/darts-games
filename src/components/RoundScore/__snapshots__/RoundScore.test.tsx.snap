// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="css-p284m"
  >
    <div
      class="css-0"
    >
      <h2
        class="chakra-heading css-82d3ew"
      >
        20T
      </h2>
    </div>
    <div
      class="css-0"
    >
      <h2
        class="chakra-heading css-82d3ew"
      >
        19T
      </h2>
    </div>
    <div
      class="css-0"
    >
      <h2
        class="chakra-heading css-82d3ew"
      >
        18T
      </h2>
    </div>
    <div
      class="css-0"
    >
      <div
        class="css-1ensw45"
      >
        <div
          class="css-0"
        >
          <button
            aria-label="clear scores"
            class="chakra-button css-b7k6lj"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M10 3h3v1h-1v9l-1 1H4l-1-1V4H2V3h3V2a1 1 0 0 1 1-1h3a1 1 0 0 1 1 1v1zM9 2H6v1h3V2zM4 13h7V4H4v9zm2-8H5v7h1V5zm1 0h1v7H7V5zm2 0h1v7H9V5z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="css-0"
        >
          <button
            class="chakra-button css-1kg18wp"
            type="button"
          >
            Round Change
          </button>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
